includes/class-newspack-popups-api.php

```diff
@@ -289,11 +291,31 @@ public function reader_post_endpoint( $request ) {
 					]
 				);
 			}
+			$this->update_cache( $transient_name, $data );
 			set_transient( $transient_name, $data, 0 );
 		}
 		return $this->reader_get_endpoint( $request );
 	}
 
+	/**
+	 * Update the cache after setting a transient.
+	 *
+	 * @param string $transient_name The transient name.
+	 * @param mixed  $data The transient data.
+	 */
+	public function update_cache( $transient_name, $data ) {
```

Comment:
Do I understand correctly, that cache is used as a parallel method of saving data to the transients stored in options table?

Reply:
Correct. The implementation is based closely on how options are handled in WordPress core. 

https://github.com/WordPress/WordPress/blob/master/wp-includes/option.php#L30-L179
https://github.com/WordPress/WordPress/blob/master/wp-includes/option.php#L331-L488